mods/sethome/init.lua

```diff
@@ -2,64 +2,55 @@ local homes_file = minetest.get_worldpath() .. "/homes"
 local homepos = {}
 
 local function loadhomes()
-    local input = io.open(homes_file, "r")
-    if input then
+	local input,err = io.open(homes_file, "r")
+	if input then
 		repeat
-            local x = input:read("*n")
-            if x == nil then
-            	break
-            end
-            local y = input:read("*n")
-            local z = input:read("*n")
-            local name = input:read("*l")
-            homepos[name:sub(2)] = {x = x, y = y, z = z}
-        until input:read(0) == nil
-        io.close(input)
-    else
-        homepos = {}
-    end
+			local x = input:read("*n")
+			if x == nil then
+				break
+			end
+			local y = input:read("*n")
+			local z = input:read("*n")
+			local name = input:read("*l")
+			homepos[name:sub(2)] = {x = x, y = y, z = z}
+		until input:read(0) == nil
+		io.close(input)
+	else
+		error(err)
+	end
 end
 
 loadhomes()
 
 minetest.register_privilege("home", "Can use /sethome and /home")
 
-local changed = false
-
 minetest.register_chatcommand("home", {
     description = "Teleport you to your home point",
     privs = {home=true},
-    func = function(name)
-        local player = minetest.get_player_by_name(name)
-        if player == nil then
-            -- just a check to prevent the server crashing
-            return false
-        end
-        if homepos[player:get_player_name()] then
-            player:setpos(homepos[player:get_player_name()])
-            minetest.chat_send_player(name, "Teleported to home!")
-        else
-            minetest.chat_send_player(name, "Set a home using /sethome")
-        end
-    end,
+	func = function(name)
+		local player = minetest.env:get_player_by_name(name)
+		if homepos[name] and player then
```

Comment:
i think it would be better if it check if the player isn't nil first then the homepos
ie: `if player and homepos[name] then`.